test/recording-history.js

```diff
@@ -0,0 +1,298 @@
+'use strict';
+
+const {serial: test} = require('ava');
+const tempy = require('tempy');
+const fs = require('fs');
+const sinon = require('sinon');
+const path = require('path');
+
+const {mockImport} = require('./helpers/mocks');
+
+const {openEditorWindow} = mockImport('./editor', 'editor');
+const plugins = mockImport('./common/plugins', 'plugins');
+const dialog = mockImport('./dialog', 'dialog');
+const Sentry = mockImport('./utils/sentry', 'sentry');
+
+const {shell} = require('electron');
+
+const {
+  recordingHistory,
+  getPastRecordings,
+  checkForActiveRecording,
+  addRecording,
+  setCurrentRecording,
+  updatePluginState,
+  stopCurrentRecording,
+  cleanPastRecordings
+} = require('../main/recording-history');
+
+const incomplete = path.resolve(__dirname, 'fixtures', 'incomplete.mp4');
+const corrupt = path.resolve(__dirname, 'fixtures', 'corrupt.mp4');
+
+test.before(t => {
+  t.context.now = new Date('2020-07-21T15:27:26.564Z');
+  t.context.clock = sinon.useFakeTimers(t.context.now.getTime());
+});
+
+test.after(t => {
+  if (t.context.clock) {
+    t.context.clock.restore();
+  }
+});
+
+test.beforeEach(() => {
+  recordingHistory.clear();
+});
+
+test.afterEach.always(t => {
+  if (t.context.paths) {
+    for (const path of t.context.paths) {
+      if (fs.existsSync(path)) {
+        fs.unlinkSync(path);
+      }
+    }
+  }
+});
+
+test('getPastRecordings', t => {
+  const existingPath = tempy.file({extension: 'mp4'});
+  const missingPath = tempy.file({extension: 'mp4'});
+
+  fs.writeFileSync(existingPath, 'data');
+  t.context.paths = [existingPath];
+
+  recordingHistory.set('recordings', [{filePath: existingPath}, {filePath: missingPath}]);
+  // RecordingHistory.mockGet('recordings', [{filePath: existingPath}, {filePath: missingPath}]);
+
+  t.deepEqual(getPastRecordings(), [{filePath: existingPath}]);
+  // T.true(recordingHistory.set.calledOnceWith('recordings', [{filePath: existingPath}]));
```

Comment:
In case you didn't intend to leave this here.

Reply:
Oops, yeah that is old. I started by mocking electron-store, but I figured why not just use it in the testing